### ISSUE TITLE
Cannot create WCSAxes from NDCube WCS

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/ndcube#214]
## Description

Per the [astropy wcsaxes](https://docs.astropy.org/en/stable/visualization/wcsaxes/), I would expect to be able to construct a `WCSAxes` instance directly from the `.wcs` attribute of an `NDCube`. However, this raises a `ValueError`.

## Expected Output

Correct `WCSAxes` instance derived from the `.wcs` attribute on the NDCube

## Actual Output

```python-traceback
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-65-89a59595f583> in <module>
      1 fig = plt.figure()
----> 2 ax = fig.gca(projection=cube_slice.wcs)
      3 cube_slice.plot(axes=ax)

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/matplotlib/figure.py in gca(self, **kwargs)
   1930 
   1931         # no axes found, so create one which spans the figure
-> 1932         return self.add_subplot(1, 1, 1, **kwargs)
   1933 
   1934     def sca(self, a):

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/matplotlib/figure.py in add_subplot(self, *args, **kwargs)
   1412                     self._axstack.remove(ax)
   1413 
-> 1414             a = subplot_class_factory(projection_class)(self, *args, **kwargs)
   1415 
   1416         return self._add_axes_internal(key, a)

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/matplotlib/axes/_subplots.py in __init__(self, fig, *args, **kwargs)
     67 
     68         # _axes_class is set in the subplot_class_factory
---> 69         self._axes_class.__init__(self, fig, self.figbox, **kwargs)
     70         # add a layout box to this, for both the full axis, and the poss
     71         # of the axis.  We need both because the axes may become smaller

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/visualization/wcsaxes/core.py in __init__(self, fig, rect, wcs, transform, coord_meta, transData, slices, frame_class, **kwargs)
    118         self._display_coords_index = 0
    119         fig.canvas.mpl_connect('key_press_event', self._set_cursor_prefs)
--> 120         self.patch = self.coords.frame.patch
    121         self._wcsaxesartist = _WCSAxesArtist()
    122         self.add_artist(self._wcsaxesartist)

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/visualization/wcsaxes/frame.py in patch(self)
    138     @property
    139     def patch(self):
--> 140         self._update_patch_path()
    141         return PathPatch(self._path, transform=self.parent_axes.transData,
    142                          facecolor=rcParams['axes.facecolor'], edgecolor='white')

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/visualization/wcsaxes/frame.py in _update_patch_path(self)
    124     def _update_patch_path(self):
    125 
--> 126         self.update_spines()
    127         x, y = [], []
    128         for axis in self:

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/visualization/wcsaxes/frame.py in update_spines(self)
    210         ymin, ymax = self.parent_axes.get_ylim()
    211 
--> 212         self['b'].data = np.array(([xmin, ymin], [xmax, ymin]))
    213         self['r'].data = np.array(([xmax, ymin], [xmax, ymax]))
    214         self['t'].data = np.array(([xmax, ymax], [xmin, ymax]))

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/visualization/wcsaxes/frame.py in data(self, value)
     46             self._pixel = self.parent_axes.transData.transform(self._data)
     47             with np.errstate(invalid='ignore'):
---> 48                 self._world = self.transform.transform(self._data)
     49             self._update_normal()
     50 

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/visualization/wcsaxes/transforms.py in transform(self, pixel)
    167             world = np.zeros((0, 2))
    168         else:
--> 169             world = self.wcs.wcs_pix2world(pixel_full, 1)
    170 
    171         # At the moment, one has to manually check that the transformation

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/wcs/wcs.py in wcs_pix2world(self, *args, **kwargs)
   1388         return self._array_converter(
   1389             lambda xy, o: self.wcs.p2s(xy, o)['world'],
-> 1390             'output', *args, **kwargs)
   1391     wcs_pix2world.__doc__ = """
   1392         Transforms pixel coordinates to world coordinates by doing

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/wcs/wcs.py in _array_converter(self, func, sky, ra_dec_order, *args)
   1290             if xy.shape == () or len(xy.shape) == 1:
   1291                 return _return_list_of_arrays([xy], origin)
-> 1292             return _return_single_array(xy, origin)
   1293 
   1294         elif len(args) == self.naxis + 1:

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/wcs/wcs.py in _return_single_array(xy, origin)
   1269                 raise ValueError(
   1270                     "When providing two arguments, the array must be "
-> 1271                     "of shape (N, {0})".format(self.naxis))
   1272             if 0 in xy.shape:
   1273                 return xy

ValueError: When providing two arguments, the array must be of shape (N, 3)
```

## Steps to Reproduce

```python
import numpy as np
import astropy.wcs
import ndcube
import matplotlib.pyplot as plt
data = np.ones((5, 4, 3))
wcs_input_dict = {
    'CTYPE3': 'WAVE    ', 'CUNIT3': 'Angstrom', 'CDELT3': 0.2, 'CRPIX3': 0, 'CRVAL3': 10, 'NAXIS3': 5,
    'CTYPE2': 'HPLT-TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5, 'NAXIS2': 4,
    'CTYPE1': 'HPLN-TAN', 'CUNIT1': 'deg', 'CDELT1': 0.4, 'CRPIX1': 2, 'CRVAL1': 1, 'NAXIS1': 3}
input_wcs = astropy.wcs.WCS(wcs_input_dict)
cube = ndcube.NDCube(data, input_wcs)
cube_slice = cube[:,:,0]
fig = plt.figure()
ax = fig.gca(projection=cube_slice.wcs)
cube_slice.plot(axes=ax)
```

### System Details

  - ndcube Version: 1.2.1
  - SunPy  Version: 1.0.3
  - astropy Version: 3.2.2
  - Python Version: 3.7.3
  - OS information: macOS 10.14.5 (Mojave)

**Comment by dstansby** [comment]
What does `print(cube_slice.wcs)` say? My guess is that the `.wcs` object is not a 2D one, but retains the 3rd dimension, which astropy isn't liking when it tries to use it as a projection.

**Comment by wtbarnes** [comment]
That seems accurate.
```python
>>> print(cube_slice.wcs)
```
```
WCS Transformation

This transformation has 3 pixel and 3 world dimensions

Array shape (Numpy order): (1, 471, 462)

Pixel Dim  Data size  Bounds
        0        462  None
        1        471  None
        2          1  None

World Dim  Physical Type                   Units
        0  custom:pos.helioprojective.lon  deg
        1  custom:pos.helioprojective.lat  deg
        2  None                            unknown

Correlation between pixel and world axes:

             Pixel Dim
World Dim    0    1    2
        0  yes  yes   no
        1  yes  yes   no
        2   no   no  yes
```

`NDCube` is clearly already doing this slicing under the hood though. Is there an easy way to get at the sliced WCS?

**Comment by dstansby** [comment]
You can do `.wcs.dropaxis(2)` manually, but maybe ndcube should be doing that under the hood when slicing?

**Comment by dstansby** [comment]
See https://github.com/astropy/astropy/pull/9440 for a more sensible error message.

**Comment by wtbarnes** [comment]
Ah I was not aware of `.dropaxis`. Adding `.dropaxis(2)` works and is a perfectly fine workaround.

I wonder what the best solution is long term. Maybe provide some method for generating a WCSAxes-friendly WCS? or just returning the WCSAxes itself?

**Comment by Cadair** [comment]
This problem is fixed in #169 and astropy/astropy#9067 because any sliced WCS is passable directly to WCSAxes. This isn't something we could fix easily in 1.x (but could provide a helper method I suppose).

**Comment by wtbarnes** [comment]
Eh if it is going to be fixed in 2.0, then using `.dropaxis` in the interim is fine.
